### PR TITLE
compact fold for typescript/javascript, see #3352

### LINKED ISSUE #111731: Hide FoldRegion end line
As discussed in #3352 :

### PR DESCRIPTION
**Pull request by aminroosta**
This PR fixes #111731

### CODE REVIEW

File: extensions/typescript-language-features/src/languageFeatures/folding.ts [hunk 1]
```diff
@@ -35,12 +36,16 @@ class TypeScriptFoldingProvider implements vscode.FoldingRangeProvider {
 			return;
 		}
 
-		return coalesce(response.body.map(span => this.convertOutliningSpan(span, document)));
+		const config = vscode.workspace.getConfiguration(this.modeId);
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/languageFeatures/folding.ts:39]: In the call to `getConfiguration`, also pass the document as the `scope`. This ensures that the correct configuration is used for the current file

File: extensions/typescript-language-features/package.nls.json [hunk 2]
```diff
@@ -92,6 +92,7 @@
 	"typescript.updateImportsOnFileMove.enabled.always": "Always update paths automatically.",
 	"typescript.updateImportsOnFileMove.enabled.never": "Never rename paths and don't prompt.",
 	"typescript.autoClosingTags": "Enable/disable automatic closing of JSX tags.",
+	"typescript.fold.compact": "Enabled/disable compact fold.",
```
Comment (mjbvz) [inline extensions/typescript-language-features/package.nls.json:95]: To help users discover this feature, add a description of what compact folding does. For example: `Enabled/disable compact folding. This hides the closing } of folded blocks.`

File: extensions/typescript-language-features/src/languageFeatures/folding.ts [hunk 3]
```diff
@@ -54,7 +59,7 @@ class TypeScriptFoldingProvider implements vscode.FoldingRangeProvider {
 		}
 
 		const start = range.start.line;
-		const end = this.adjustFoldingEnd(range, document);
+		const end = compact ? range.end.line : this.adjustFoldingEnd(range, document);
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/languageFeatures/folding.ts:62]: Does compact folding need to do anything special for cases where the closing `}` also has text content on the same line? For example:

```ts
if (true) {
    console.log(1); }
```

and

```ts
if (true) {
      console.log(1); 
};   console.log(2); 
```

**Review by mjbvz — CHANGES_REQUESTED** [review]
Looks good! Just a few minor changes/questions. Let's also confirm with @aeschli that the new setting names are consistent across languages

**Comment by aminroosta** [comment]
Thanks so much for the review, will fix them over the weekend.

**Comment by mjbvz** [comment]
@aminroosta Hey, just checking on on this PR. Let me know if you have any questions about my comments

File: extensions/typescript-language-features/package.nls.json (hunk 2 above)
Comment (yuzhishuo) [inline extensions/typescript-language-features/package.nls.json:95]: lgtm

**Comment by waldyrious** [comment]
> **aminroosta** closed this

🙁 What a pity. I was really looking forward to this one.

**Pull request closed without merging**